paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -100,13 +93,15 @@ internal class PaymentOptionsViewModel @Inject constructor(
 
         val linkState = args.state.linkState
 
-        _isLinkEnabled.value = linkState != null
-        activeLinkSession.value = linkState?.loginState == LinkState.LoginState.LoggedIn
-
-        if (linkState != null) {
-            setupLink(linkState)
+        viewModelScope.launch {
```

Comment:
I think this would make the view models harder to test, since this is causing side effects that we don't have control over.